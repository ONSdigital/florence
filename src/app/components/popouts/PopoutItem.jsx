import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";

const propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string,
    body: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    buttons: PropTypes.arrayOf(
        PropTypes.shape({
            onClick: PropTypes.func.isRequired,
            text: PropTypes.string.isRequired,
            style: PropTypes.oneOf(["standard", "positive", "primary", "warning", "subtle", "invert-primary"]),
        })
    ),
};

const PopoutItem = props => (
    <div>
        <div className="modal__header">
            <h1 className="modal__title">{props.title}</h1>
        </div>
        <div className="modal__body">
            {props.body}
            <div className="modal__button-container">
                {props.buttons.map((button, i) => {
                    const buttonClasses = clsx("btn", "btn--margin-right", {
                        [`btn--${button.style}`]: button.style && button.style !== "standard",
                    });
                    return (
                        <button key={`button--${i}`} className={buttonClasses} onClick={button.onClick}>
                            {button.text}
                        </button>
                    );
                })}
            </div>
        </div>
    </div>
);

PopoutItem.propTypes = propTypes;
export default PopoutItem;
