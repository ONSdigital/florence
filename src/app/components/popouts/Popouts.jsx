import React from "react";
import PropTypes from "prop-types";
import PopoutItem from "./PopoutItem";
import Modal from "../Modal";

const propTypes = {
    popouts: PropTypes.arrayOf(PropTypes.shape(PopoutItem.propTypes)).isRequired,
};

const Popouts = props => {
    if (props.popouts.length === 0) return null;
    ``;

    const modalChild = (
        <div className="">
            {props.popouts.map(popout => {
                return <PopoutItem {...popout} key={popout.id} />;
            })}
        </div>
    );
    return <Modal children={modalChild} sizeClass="grid__col-4" />;
};

Popouts.propTypes = propTypes;
export default Popouts;
