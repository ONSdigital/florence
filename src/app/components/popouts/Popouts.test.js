import React from "react";
import Popouts from "./Popouts";
import PopoutItem from "./PopoutItem.jsx";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

describe("Popouts", () => {
    describe("when empty popouts array given as prop", () => {
        it("do not render", () => {
            const component = mount(<Popouts popouts={[]} />);
            expect(component.isEmptyRender()).toBe(true);
        });
    });

    describe("when popouts props given to Popout", () => {
        const popouts = [
            {
                id: "foo",
                title: "bar",
                body: "baz",
                buttons: [
                    {
                        onClick: () => {},
                        text: "Qux",
                        style: "primary",
                    },
                ],
            },
        ];

        const component = mount(<Popouts popouts={popouts} />);

        it("it renders PopoutItem", () => {
            expect(component.find("h1").hasClass("modal__title")).toBe(true);
            expect(component.find(PopoutItem)).toHaveLength(popouts.length);
        });
        it("matches the snapshot", () => {
            const componentForSnapshot = renderer.create(<Popouts popouts={popouts} />);
            expect(componentForSnapshot.toJSON()).toMatchSnapshot();
        });
    });
});
