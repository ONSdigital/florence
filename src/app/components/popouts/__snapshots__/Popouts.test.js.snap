// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popouts when popout props Popouts matches the snapshot 1`] = `
<div
  className="modal__overlay"
>
  <div
    className="modal__inner grid__col-4"
  >
    <div
      className=""
    >
      <div>
        <div
          className="modal__header"
        >
          <h1
            className="modal__title"
          >
            bar
          </h1>
        </div>
        <div
          className="modal__body"
        >
          baz
          <div
            className="modal__button-container"
          >
            <button
              className="btn btn--margin-right btn--primary"
              onClick={[Function]}
            >
              Qux
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
